uptrain/operators/language/generation.py

```diff
@@ -170,6 +171,104 @@ def run(self, data: pl.DataFrame) -> TYPE_TABLE_OUTPUT:
         }
 
 
+@register_op
+class TopicGenerator(ColumnOp):
+    """
+    Takes a table of clustered texts and identifies the topic for each cluster 
+
+    Attributes:
+        col_in_cluster_index (str): The name of the column containing the cluster index.
+        col_in_embs (str): The name of the column containing the embeddings.
+        col_in_text (str): The name of the column containing the text where the grouping needs to be performed.
+        col_out_text (str): The name of the column containing the topic for each entry.
+        temperature (float): Temperature for the LLM to generate responses.
+        model (str): which LLM to use for topic generator.
+
+    Returns:
+        TYPE_TABLE_OUTPUT: A dictionary containing the dataset with the output text.
+    """
+    col_in_cluster_index: str = 'cluster_index'
+    col_in_text: str = 'question'
+    col_in_embs: str = 'umap_embedding'
+    col_out_text: str = 'topic'
+    model: str = 'gpt-3.5-turbo'
+    temperature: float = 1.0
+    _api_client: LLMMulticlient
+
+    def setup(self, settings: Settings):
+        self._api_client = LLMMulticlient(settings=settings)
+        return self
+
+    def _make_payload(self, id: t.Any, text: str, model: str) -> Payload:
+        return Payload(
+            data={
+                "model": model,
+                "messages": [{"role": "user", "content": text}],
+                "temperature": self.temperature,
+            },
+            metadata={"index": id},
+        )
+
+    def run(self, data: pl.DataFrame) -> TYPE_TABLE_OUTPUT:
+
+        embeddings = np.asarray(list(data[self.col_in_embs]))
+        questions = np.asarray(data[self.col_in_question])
+        n_clusters = max(data[self.cluster_index])
+
+        input_payloads = []
+        outputs = [None] * len(questions)
+        indexes_cluster = []
+        
+        for index in range(n_clusters+1):
+            points = data[self.cluster_index]==index
+            indexes_cluster.append(list(np.where(points)[0]))
+            embeddings_cluster = embeddings[points]
+            questions_cluster = questions[points]
+            centroid = np.mean(embeddings_cluster, axis=0)
+            indexes_sorted = np.argsort(np.linalg.norm(embeddings_cluster - centroid, axis=1))
```

Comment:
This might lead to duplicates. Should we do sampling? @ashish-1600

Reply:
Sampling might give too much variability. Should we take the top ones which are unique? @sourabhagr

Reply:
I guess we can allow both. My thoughts:
1. Top K uniques -> We might end up with very similar-looking candidates and that might not reflect the true topic of the cluster
2. Sampling -> Agreed, can bring variability across runs. 

Will raise an issue for this.